src/nvapi.cpp

```diff
@@ -15,7 +15,8 @@ extern "C" {
     NvAPI_Status __cdecl NvAPI_EnumLogicalGPUs(NvLogicalGpuHandle nvGPUHandle[NVAPI_MAX_LOGICAL_GPUS], NvU32* pGpuCount) {
         constexpr auto n = __func__;
 
-        Enter(n);
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(nvGPUHandle), log::fmt::ptr(pGpuCount));
```

Comment:
The idea behind was to not execute all those `log::fmt` functions with the string operations inside when not tracing since some of our functions  might be on a hot path.

Reply:
Hmm, fair. One could use macros for this but if you don't want to bring them here then this is probably the best we can do.